app.js

```diff
@@ -30,13 +34,96 @@ app.get('/vegetables/:id', (req, res, next) => {
   res.json(vegetable)
 })
 
+app.delete('/vegetables/:id', (req, res, next) => {
+  const { vegetables } = data;
+  const { id } = req.params;
+  const updatedVegetables = vegetables.filter(veggie => !veggie.id === id);
```

Comment:
`.filter()` will not modify an array permanently. Instead, it will return a new array with the filtered items and leave the original array intact. `.splice()` would probably be a better method here.